test/unit/org/apache/cassandra/index/sai/cql/GenericOrderByTest.java

```diff
@@ -114,7 +114,7 @@ public void testPrimaryKeyRestrictionToEnsureBoundsAreCorrectlyHandled() throws
 
         // Insert many rows and then ensure we can get each of them when querying with specific bounds.
         for (int i = 0; i < 100; i++)
-            execute("INSERT INTO %s (pk, x, val, str_val) VALUES (?, ?, ?, ?)", i, i, i, i);
+            execute("INSERT INTO %s (pk, x, val, str_val) VALUES (?, ?, ?, ?)", i, i, i, String.valueOf(i));
```

Comment:
This looks like an example of how `executeInternal` misses some validation done by `execute`.

Reply:
Hmm interesting, the test passes for me even without the change?

Reply:
Right, this is a mistake due to issues that I had with the implementation of `executeFormattedQuery`. I'll revert.